lms/views/helpers/_via.py

```diff
@@ -1,8 +1,82 @@
 """Via-related view helpers."""
+import re
 from urllib.parse import parse_qsl, urlencode, urlparse
 
 __all__ = ["via_url"]
 
+# pylint: disable=too-few-public-methods
+
+
+class _ViaDoc:
+    """A doc we want to proxy with content type."""
+
+    GOOGLE_DRIVE_REGEX = re.compile(
+        r"^https://drive.google.com/uc\?id=(.*)&export=download$", re.IGNORECASE
+    )
+
+    def __init__(self, url, content_type=None):
+        self.url = url
+
+        if content_type is None and self.GOOGLE_DRIVE_REGEX.match(url):
+            content_type = "pdf"
+
+        self._content_type = content_type
+
+    @property
+    def is_pdf(self):
+        return self._content_type == "pdf"
+
+    @property
+    def is_html(self):
+        return self._content_type == "html"
```

Comment:
I know it's a bit _extra_, but go with me. I wanted the core function below to be very declarative.

The idea is we can also specify the content type if we know it when we create this object and the right thing (tm) should happen as a result.

Reply:
Note also, that `is_pdf() != not is_html()`. If we have a `content_type` of `None` we don't know what it is, and will have to route to find out.